eland/operations.py

```diff
@@ -145,51 +145,73 @@ def count(self, query_compiler):
-        return build_pd_series(results, index=results.keys())
+        return build_pd_series(
+            results, index=results.keys(), dtype=(np.float64 if numeric_only else None)
+        )
 
-    def median(self, query_compiler, numeric_only=True):
+    def median(self, query_compiler, numeric_only: Optional[bool] = None):
         results = self._metric_aggs(
             query_compiler, ["median"], numeric_only=numeric_only
         )
-        return build_pd_series(results, index=results.keys())
+        return build_pd_series(
+            results, index=results.keys(), dtype=(np.float64 if numeric_only else None)
+        )
 
-    def sum(self, query_compiler, numeric_only=True):
+    def sum(self, query_compiler, numeric_only: Optional[bool] = None):
         results = self._metric_aggs(query_compiler, ["sum"], numeric_only=numeric_only)
-        return build_pd_series(results, index=results.keys())
+        return build_pd_series(
+            results, index=results.keys(), dtype=(np.float64 if numeric_only else None)
+        )
 
-    def max(self, query_compiler, numeric_only=True):
+    def max(self, query_compiler, numeric_only: Optional[bool] = None):
         results = self._metric_aggs(query_compiler, ["max"], numeric_only=numeric_only)
-        return build_pd_series(results, index=results.keys())
+        return build_pd_series(
+            results, index=results.keys(), dtype=(np.float64 if numeric_only else None)
+        )
 
-    def min(self, query_compiler, numeric_only=True):
+    def min(self, query_compiler, numeric_only: Optional[bool] = None):
         results = self._metric_aggs(query_compiler, ["min"], numeric_only=numeric_only)
-        return build_pd_series(results, index=results.keys())
+        return build_pd_series(
+            results, index=results.keys(), dtype=(np.float64 if numeric_only else None)
+        )
 
     def nunique(self, query_compiler):
         results = self._metric_aggs(query_compiler, ["nunique"], numeric_only=False)
         return build_pd_series(results, index=results.keys())
 
-    def mad(self, query_compiler, numeric_only=True):
+    def mad(self, query_compiler, numeric_only: Optional[bool] = None):
         results = self._metric_aggs(query_compiler, ["mad"], numeric_only=numeric_only)
-        return build_pd_series(results, index=results.keys())
+        return build_pd_series(
+            results, index=results.keys(), dtype=(np.float64 if numeric_only else None)
+        )
 
     def value_counts(self, query_compiler, es_size):
         return self._terms_aggs(query_compiler, "terms", es_size)
 
     def hist(self, query_compiler, bins):
         return self._hist_aggs(query_compiler, bins)
 
-    def _metric_aggs(self, query_compiler: "QueryCompiler", pd_aggs, numeric_only=True):
+    def _metric_aggs(
+        self,
+        query_compiler: "QueryCompiler",
+        pd_aggs,
+        numeric_only: Optional[bool] = None,
+        is_aggregation: bool = False,
```

Comment:
- Because we use same method `_metric_aggs` for both agg and non-aggs. Both will pass an agg as list.
- `df.mean()` => will pass ["mean"] to `_metric_aggs`
- `df.agg(["mean"])` => will do the same as above
- And we have different case for numeric_only for aggs and non-aggs.
To differentiate them I introduced the `is_aggregation`

Reply:
Yes, Will rename it.